### ISSUE TITLE
AiModelResult.Predict() returns all-zero output for BinaryClassification after successful training

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1625]
## Summary
After training a Dense MLP via the `AiModelBuilder` facade with `NeuralNetworkTaskType.BinaryClassification` and `BinaryCrossEntropyLoss`, the resulting `AiModelResult.Predict(testX)` returns a tensor of all-zero values — for every input, regardless of training success. The same happens when calling `AiModelBuilder.Predict(testX, result)`.

This is reproducible with the canonical facade walkthrough pattern. Verified on AiDotNet 0.159.0.

## Repro

```csharp
// Setup: 800 train, 200 test, 9 features, roughly balanced binary labels
// Model: DenseLayer(16, ReLU) -> DenseLayer(8, ReLU) -> DenseLayer(1, Sigmoid)
// Optimizer: Adam, lr=0.02, MaxIterations=150, BatchSize=64
// Loss: BinaryCrossEntropyLoss
// TaskType: BinaryClassification

var nn = new NeuralNetwork<double>(architecture);
var builder = new AiModelBuilder<double, Tensor<double>, Tensor<double>>();
var model = await builder
    .ConfigureModel(nn)
    .ConfigureOptimizer(adam)
    .ConfigureLossFunction(new BinaryCrossEntropyLoss<double>())
    .ConfigureDataLoader(new InMemoryDataLoader<double, Tensor<double>, Tensor<double>>(trainX, trainY))
    .BuildAsync();

var preds = builder.Predict(testX, model);
// Observed: preds[i, 0] == 0.0 (exactly) for ALL i — even with :G17 format

// Diagnostic: the underlying NN's raw forward DOES produce varied probabilities
var rawProbs = nn.ForwardForTraining(testX);   // 0.16, 0.52, 0.39, 0.33, 0.23, ...
```

## Expected
`Predict()` for `BinaryClassification` should return either:
- The raw sigmoid probabilities (so callers can choose their own threshold), OR
- Binary 0/1 labels derived from a documented threshold (e.g., 0.5)

Either way, when the underlying model produces values both above and below the natural 0.5 boundary, `Predict()` should reflect that — not collapse everything to 0.

## Actual
`Predict()` returns exactly 0.0 for every input, even when `nn.ForwardForTraining(testX)` on the same trained `nn` shows probabilities spanning ~0.15 to ~0.52 — meaning at least some samples should binarize to 1 if the threshold is 0.5.

## Impact
This blocks:
- **Fairness audits** (W9 Ethics & RAI curriculum) — need per-group selection rates to be non-trivial
- **Threshold tuning** — can't choose a non-0.5 threshold if `Predict` already binarizes
- **ROC curves / AUC** — require probabilities
- **Calibration analysis** — requires probabilities
- **Production decision logic** — can't apply business-rule thresholds

The facade walkthrough examples (1, 3, 4, 5) all exhibit this — their "73% / 86% / 73% accuracy with all-0.00 sample predictions" is the test set's negative-class base rate, not actual learning visible through the facade.

## Workaround (currently)
Drop out of the facade into the NN directly:
```csharp
nn.SetTrainingMode(false);
var probs = nn.ForwardForTraining(testX);  // varied probabilities
```
But this bypasses the documented facade pattern and is brittle.

## Suggested Fix
1. Either: make `Predict()` return probabilities for BinaryClassification (most flexible), OR
2. Add a new facade method like `AiModelResult.PredictProbabilities(TInput)` or `AiModelResult.PredictWithScores(TInput)` that returns probabilities while leaving `Predict()` as the binarized label
3. Document the threshold used by `Predict()` for BinaryClassification (currently undocumented, and observed behavior suggests it's not the natural 0.5)

## Environment
- AiDotNet 0.159.0
- .NET 10.0
- Windows 11
- Reproducible across multiple optimizer settings (lr 0.001 to 0.05, MaxIterations 20 to 500, BatchSize 32 to 800)
